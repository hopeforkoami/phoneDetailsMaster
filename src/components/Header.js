import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import GoBackIcon from '../assets/GoBackIcon.png';
import MicIcon from '../assets/MicIcon.png';
import SettingsIcon from '../assets/SettingsIcon.png';

const Header = (props) => {
  const { previewLink } = props;
  return (<header>
      <nav>
        <button type="button" className="navigation-icons"><Link to={previewLink}><img src={GoBackIcon} alt="Go back icon" /></Link></button>
        <h1>Phone Details Master</h1>
        <div>
          <button type="button" className="navigation-icons"><img src={MicIcon} alt="Go back icon" /></button>
          <button type="button" className="navigation-icons"><img src={SettingsIcon} alt="Go back icon" /></button>
        </div>
      </nav>
    </header>
  );
};

export default Header;

Header.propTypes = {
  previewLink: PropTypes.string,
};