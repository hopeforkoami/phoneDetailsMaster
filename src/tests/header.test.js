import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import Header from '../components/Header';

test('renders Header component', () => {
  const { asFragment } = render(
    <Router>
      <Header />
    </Router>,
  );
  expect(asFragment()).toMatchSnapshot();
});

test('renders Header component', () => {
  const { getByText } = render(
    <Router>
      <Header />
    </Router>,
  );
  expect(getByText('Phone Details Master')).toBeInTheDocument();
});