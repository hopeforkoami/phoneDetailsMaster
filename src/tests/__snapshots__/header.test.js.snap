// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header component 1`] = `
<DocumentFragment>
  <header>
    <nav>
      <button
        class="navigation-icons"
        type="button"
      >
        <a
          href="/"
        >
          <img
            alt="Go back icon"
            src="GoBackIcon.png"
          />
        </a>
      </button>
      <h1>
        Phone Details Master
      </h1>
      <div>
        <button
          class="navigation-icons"
          type="button"
        >
          <img
            alt="Go back icon"
            src="MicIcon.png"
          />
        </button>
        <button
          class="navigation-icons"
          type="button"
        >
          <img
            alt="Go back icon"
            src="SettingsIcon.png"
          />
        </button>
      </div>
    </nav>
  </header>
</DocumentFragment>
`;
